### ISSUE TITLE
Time based notification system for attendance

### ISSUE THREAD

**Issue by sjain30** [collegeconnect/CollegeConnect#50]
Implement a time-based notification system for attendance by scheduling notification at the end of class, reminding the user to mark their attendance in the app. Details such as subject name and end time are stored in the timetable database.

**Comment by sakshampruthi** [comment]
Try and use work manager for triggering notification. Incase the timings are updated cancel the previous job and initialize a new for the new input time

**Comment by AvishekThapa1997** [comment]
Is this issue still open?

**Comment by sjain30** [comment]
@AvishekThapa1997 Yes, it's open. If you wanna work on it let us know, we'll assign you to it.

**Comment by AvishekThapa1997** [comment]
Yes i will work.

**Comment by AvishekThapa1997** [comment]
user will get the notification only when the app is on the screen or even in during when the app is in foreground?

**Comment by sakshampruthi** [comment]
Every possible scenario on screen, in foreground or in background

**Comment by AvishekThapa1997** [comment]
there is no google-services.json file due to which app is not getting build.Any help would be appreciated.

**Comment by sjain30** [comment]
@AvishekThapa1997 we can't share aur google services file due to security reasons. You would have to link it to your own firebase project

**Comment by sakshampruthi** [comment]
@AvishekThapa1997 Please take look at [Contributing guideline](/collegeconnect/CollegeConnect/blob/development/CONTRIBUTING.md)

**Comment by AvishekThapa1997** [comment]
Okay ...this is my first time i am trying open source so getting lots of questions

**Comment by sakshampruthi** [comment]
@AvishekThapa1997 No problem, connect with us on gitter for any further questions [![Gitter](https://badges.gitter.im/CollegeConnect/discussion.svg)](https://gitter.im/CollegeConnect/discussion?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)

**Comment by sakshampruthi** [comment]
@AvishekThapa1997 Are you still working on this issue?

**Comment by AvishekThapa1997** [comment]
Assign it to someone else.

**Comment by rooparsh** [comment]
@sakshampruthi Can you assign this issue to me, am looking into it .

**Comment by sakshampruthi** [comment]
Hi @rooparsh could you describe your approach?

**Comment by rooparsh** [comment]
Using the Endtime store in db, will schedule a notification using work manager.

**Comment by sakshampruthi** [comment]
Work manager is generally used for deferred task and is not reliable to trigger the notification at the required time becuase of doze. It will be scheduled according to the phone maintenance time. I would suggest you to develop a solution using alarm manager and and every time its triggered, access room db with power manager and schedule another alarm using setExactAndAllowWhileIdle()

**Comment by rooparsh** [comment]
@sakshampruthi We can use periodic work request. I am assuming the classes will be 1 hr long each. So after every 1 hr, a notification can be triggered. And at the end of the last period, we can cancel the job and schedule a new job for next day. 
Assign this issue to me, will raise a PR , we can discuss the approach there.

**Comment by zubusoomro** [comment]
is this still open ?

**Comment by princessdharmy** [comment]
@sakshampruthi is this still open to work on?
